03_analysis/compare_ffs.py

```diff
@@ -0,0 +1,445 @@
+
+            # initialize lists to store conformer energies
+            enes_ref = []
+            enes_que = []
+            rmsds_mol = []
+            tfds_mol = []
+
+            # loop over each conformer of this mol
+            for ref_conf, que_conf in zip(rmol.GetConfs(), qmol.GetConfs()):
+
+                # check confomer match from the specified tag
+                ref_id = oechem.OEGetSDData(ref_conf, conf_id_tag)
+                que_id = oechem.OEGetSDData(que_conf, conf_id_tag)
+                if ref_id != que_id:
+                    raise ValueError("ERROR: Conformers not aligned in iteration"
+                                    f" for mol: '{rmol_name}'. The conformer "
+                                    f"IDs ({conf_id_tag}) for ref and query are:"
+                                    f"\n{ref_id}\n{que_id}.")
+
+                # get energies
+                enes_ref.append(float(oechem.OEGetSDData(ref_conf, tag_ref)))
+                enes_que.append(float(oechem.OEGetSDData(que_conf, tag_que)))
+
+                # compute RMSD between reference and query conformers
+                rmsd = oechem.OERMSD(ref_conf, que_conf, automorph,
+                                     heavyOnly, overlay)
+                rmsds_mol.append(rmsd)
+
+                # compute TFD between reference and query conformers
+                tfd = calc_tfd(ref_conf, que_conf)
+                tfds_mol.append(tfd)
+
+                # store data in SD tags for query conf, and write conf to file
+                oechem.OEAddSDData(que_conf, f'RMSD to {sdf_ref}', str(rmsd))
+                oechem.OEAddSDData(que_conf, f'TFD to {sdf_ref}', str(tfd))
+                oechem.OEWriteConstMolecule(ofs, que_conf)
+
+            # compute relative energies against lowest E reference conformer
+            lowest_ref_idx = enes_ref.index(min(enes_ref))
+            rel_enes_ref = np.array(enes_ref) - enes_ref[lowest_ref_idx]
+            rel_enes_que = np.array(enes_que) - enes_que[lowest_ref_idx]
+
+            # subtract them to get ddE = dE (ref method) - dE (query method)
+            enes_mol = np.array(rel_enes_ref) - np.array(rel_enes_que)
+
+            # store then move on
+            enes_method.append(enes_mol)
+            rmsds_method.append(np.array(rmsds_mol))
+            tfds_method.append(np.array(tfds_mol))
+            #print(rmsds_method, len(rmsds_method))
+            #print(enes_method, len(enes_method))
+
+        enes_full.append(enes_method)
+        rmsds_full.append(rmsds_method)
+        tfds_full.append(tfds_method)
+
+    ofs.close()
+
+    return enes_full, rmsds_full, tfds_full
+
+
+def flatten(list_of_lists):
+    """
+    Flatten one level of nesting.
+
+    Parameter
+    ---------
+    list_of_lists
+
+    Returns
+    -------
+    1D numpy array
+
+    """
+    return np.concatenate(list_of_lists).ravel()
+
+
+def draw_scatter(xdata, ydata, method_labels, xlabel, ylabel, outfile):
```

Comment:
One tip: When wiring in font sizes, etc., also specify the SIZE of the figure. Otherwise if you, say, get the figure output at two-column size and then rescale it to a single column, the font size will end up too small for a journal.

You might also want a flag for "talk" vs "paper" axis labels -- in a talk you often want bigger axis labels than in a paper.

(These are just general tips on making it easy to reuse plotting code for multiple purposes without having to change it a ton of times.)

Reply:
(So the same comments would apply to ridgeplots, etc.)